### ISSUE TITLE
bcbio_vm.py import error

### ISSUE THREAD

**Issue by matthdsm** [bcbio/bcbio-nextgen#2484]
Hi Brad,

I updated one of our test installs to the latest stable version of bcbio. Next I installed the bcbio-nextgen-vm package using conda
```
conda install -f -c conda-forge -c bioconda bcbio-nextgen-vm
```

When I try to generate the CWL workflows I get an `ImportError` for the Ansible `callbacks` module

```
Traceback (most recent call last):
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/bin/bcbio_vm.py", line 4, in <module>
    __import__('pkg_resources').run_script('bcbio-nextgen-vm==0.1.0a0', 'bcbio_vm.py')
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 658, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1445, in run_script
    exec(script_code, namespace, namespace)
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/bcbio_nextgen_vm-0.1.0a0-py2.7.egg/EGG-INFO/scripts/bcbio_vm.py", line 27, in <module>
    
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/bcbio_nextgen_vm-0.1.0a0-py2.7.egg/bcbiovm/aws/cluster.py", line 8, in <module>
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/bcbio_nextgen_vm-0.1.0a0-py2.7.egg/bcbiovm/aws/bootstrap.py", line 7, in <module>
  File "/user/data/gent/gvo000/gvo00082/bcbio/anaconda/lib/python2.7/site-packages/bcbio_nextgen_vm-0.1.0a0-py2.7.egg/bcbiovm/aws/common.py", line 9, in <module>
ImportError: No module named callbacks
```

Installing ansible from conda doesn't resolve the issue. Do you have any further tips?

Thanks
M

**Comment by chapmanb** [comment]
Matthias;
Sorry about the issue. It appears you might have somehow gotten an old version of bcbio-nextgen-vm. What version do you get when you do:
```
$ conda list | grep bcbio-nextgen-vm
bcbio-nextgen-vm          0.1.0a                 py27_125    bioconda
```
You should have py27_125 for the latest which won't have this issue since it removes the ansible imports.

I'd suggest installing bcbio-nextgen-vm in an isolated conda environment instead of with your bcbio-nextgen install. It's independent and also may be impacted by all the dependencies in that main install. Hope this helps explain what is going on.

**Comment by matthdsm** [comment]
Hi Brad,

Thanks for the tip. I assumed I had the latest version, but there were some dependency conflicts in the bcbio conda environment. Forcing an install of the latest version fixed the issue.

Thanks for the help!

Just FYI, conda reported the following conflicts:
```
-bash-4.2$ conda install -f -c conda-forge -c bioconda bcbio-nextgen-vm=0.1.0a=py27_125
Solving environment: done

## Package Plan ##

  added / updated specs:
    - bcbio-nextgen-vm==0.1.0a=py27_125


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    extract-sv-reads-1.2.1     |                0         918 KB  bioconda
    samtools-1.7               |                2         1.7 MB  bioconda
    curl-7.45.0                |                0         528 KB
    arvados-python-client-0.1.20171211211613|           py27_1         309 KB  bioconda
    lumpy-sv-0.2.14a           |           py27_1         3.8 MB  bioconda
    verifybamid2-1.0.5         |                0        20.8 MB  bioconda
    arvados-cwl-runner-1.0.20180216164101|           py27_1          95 KB  bioconda
    pysam-0.14.1               | py27_htslib1.7_0         7.1 MB  bioconda
    bcbio-nextgen-vm-0.1.0a    |         py27_125          96 KB  bioconda
    pycurl-7.19.5.3            |           py27_0         128 KB
    ------------------------------------------------------------
                                           Total:        35.5 MB

The following packages will be UPDATED:

    arvados-cwl-runner:    1.0.20160502202716-py27_0 bioconda    --> 1.0.20180216164101-py27_1 bioconda
    arvados-python-client: 0.1.20160517202250-py27_0 bioconda    --> 0.1.20171211211613-py27_1 bioconda
    bcbio-nextgen-vm:      0.1.0a-py27_87            bioconda    --> 0.1.0a-py27_125           bioconda

The following packages will be DOWNGRADED:

    bcftools:              1.8-h4da6232_3            bioconda    --> 1.7-0                     bioconda
    curl:                  7.61.0-h93b3f91_1         conda-forge --> 7.45.0-0
    extract-sv-reads:      1.2.1-h1af57d8_1          bioconda    --> 1.2.1-0                   bioconda
    lumpy-sv:              0.2.14a-hdfb72b2_2        bioconda    --> 0.2.14a-py27_1            bioconda
    pycurl:                7.43.0.2-py27hb7f436b_0               --> 7.19.5.3-py27_0
    pysam:                 0.14.1-py27hae42fb6_1     bioconda    --> 0.14.1-py27_htslib1.7_0   bioconda
    samtools:              1.9-h46bd0b3_0            bioconda    --> 1.7-2                     bioconda
    verifybamid2:          1.0.5-h601c9b8_2          bioconda    --> 1.0.5-0                   bioconda

```

On an related note, wouldn't it be possible to include bcbio_vm into the default bcbio install? I suppose this makes sense, since the plan is to move to CWL entirely?


Thanks again
Cheers
M

**Comment by matthdsm** [comment]
fixed with c92abf8.
Thank a lot! This will make my life a bit easier.

Cheers
M

**Comment by chapmanb** [comment]
Matthias;
You are too quick for me. I also wanted to add documentation prior to responding here and closing this. It's all documented now as well: https://bcbio-nextgen.readthedocs.io/en/latest/contents/cwl.html Thanks again for the helpful feedback and let me know if you run into any problems.
